### PR TITLE
[WIP] properly name account uuid and username columns

### PR DESCRIPTION
**Pull request by butonic**
- [ ] add migration step
- [ ] chaneg ocurrences of setUserId and getUserId in the code to use the UUID or the user name. 
- [ ] generate a uuid if it isn't provided by the backend (or always generate one but allow the backend to overwrite it)

for syncing uuid should be ok ... but then again other backends may not provide a uuid ... they need to sync based on he login ... hmmm

### CODE REVIEW

File: lib/private/User/Account.php [hunk 1]
```diff
@@ -54,8 +56,8 @@ class Account extends Entity {
 	const STATE_DELETED = 3;
 
 	protected $email;
-	protected $userId;
-	protected $lowerUserId;
+	protected $uuid;
+	protected $userName;
```
Comment (DeepDiver1975) [inline lib/private/User/Account.php]: not login name?

Comment (butonic) [reply lib/private/User/Account.php]: it is more than just a login name. it is shown to other users. we could eg. use it in the sharing dialog as it identifies a user. so it is more a user name than a login name.

**Comment by felixboehm** [comment]
@butonic working on this? please add milestone.
Is there an overview for current tasks and concept related to user management?

**Comment by butonic** [comment]
superseded by https://github.com/owncloud/core/pull/31335

**Pull request closed without merging**